internal/provider/kubernetes/status.go

```diff
@@ -469,6 +469,29 @@ func (r *gatewayAPIReconciler) subscribeAndUpdateStatus(ctx context.Context, ext
 	}
 }
 
+// mergeRouteParentStatus merges the old and new RouteParentStatus.
+// This is needed because the RouteParentStatus doesn't support strategic merge patch yet.
+func mergeRouteParentStatus(ns string, old, new []gwapiv1.RouteParentStatus) []gwapiv1.RouteParentStatus {
+	merged := make([]gwapiv1.RouteParentStatus, len(old))
+	_ = copy(merged, old)
+	for _, parent := range new {
+		found := -1
+		for i, existing := range old {
+			if parent.ParentRef.Name == existing.ParentRef.Name &&
+				utils.NamespaceDerefOr(parent.ParentRef.Namespace, ns) == utils.NamespaceDerefOr(existing.ParentRef.Namespace, ns) {
```

Comment:
"k8s.io/utils/ptr"
we've updated a lot of the code in the repo to use `ptr.Deref(parent.ParentRef.Namespace, ns)` instead